### PR TITLE
[main] github actions paths-ignore

### PR DESCRIPTION
**Pull request by Ginxo**
### Referenced pull requests

* https://github.com/kiegroup/optaplanner/pull/1481
* https://github.com/kiegroup/optaplanner-quickstarts/pull/151
* https://github.com/kiegroup/optaweb-vehicle-routing/pull/540
* https://github.com/kiegroup/optaweb-employee-rostering/pull/668

### CODE REVIEW

**Review by radtriste — APPROVED** [review]
lgtm

File: .github/workflows/jpa_integration_ci.yml [hunk 1]
```diff
@@ -6,9 +6,26 @@ on:
   push:
     branches:
       - main
+    paths-ignore:
+      - 'LICENSE*'
+      - '.gitignore'
+      - '**.md'
+      - '**.adoc'
+      - '*.txt'
+      - 'build/**'
```
Comment (rsynek) [inline .github/workflows/jpa_integration_ci.yml:15]: The `build` directory contains the `optaplanner-build-parent`; the parent pom nearly every other module inherits from. That's why should cover it by CI.

This applies to every `.yml` file changed in this PR.

**Review by rsynek — CHANGES_REQUESTED** [review]

File: .github/workflows/jpa_integration_ci.yml (hunk 1 above)
Comment (Ginxo) [reply .github/workflows/jpa_integration_ci.yml:15]:
```suggestion
```

File: .github/workflows/jpa_integration_ci.yml [hunk 2]
```diff
@@ -6,9 +6,26 @@ on:
...
+    types: [opened, synchronize, reopened, ready_for_review]
     branches:
       - main
+    paths-ignore:
+      - 'LICENSE*'
+      - '.gitignore'
+      - '**.md'
+      - '**.adoc'
+      - '*.txt'
+      - 'build/**'
```
Comment (Ginxo) [inline .github/workflows/jpa_integration_ci.yml:27]:
```suggestion
```

File: .github/workflows/os_java_ci.yml [hunk 3]
```diff
@@ -6,9 +6,26 @@ on:
   push:
     branches:
       - main
+    paths-ignore:
+      - 'LICENSE*'
+      - '.gitignore'
+      - '**.md'
+      - '**.adoc'
+      - '*.txt'
+      - 'build/**'
```
Comment (Ginxo) [inline .github/workflows/os_java_ci.yml:15]:
```suggestion
```

File: .github/workflows/os_java_ci.yml [hunk 4]
```diff
@@ -6,9 +6,26 @@ on:
...
+    types: [opened, synchronize, reopened, ready_for_review]
     branches:
       - main
+    paths-ignore:
+      - 'LICENSE*'
+      - '.gitignore'
+      - '**.md'
+      - '**.adoc'
+      - '*.txt'
+      - 'build/**'
```
Comment (Ginxo) [inline .github/workflows/os_java_ci.yml:27]:
```suggestion
```

File: .github/workflows/sonarcloud.yml [hunk 5]
```diff
@@ -5,6 +5,14 @@ on:
   push:
     branches:
       - main
+    paths-ignore:
+      - 'LICENSE*'
+      - '.gitignore'
+      - '**.md'
+      - '**.adoc'
+      - '*.txt'
+      - 'build/**'
```
Comment (Ginxo) [inline .github/workflows/sonarcloud.yml:14]:
```suggestion
```

File: .github/workflows/jpa_integration_ci.yml (hunk 1 above)
Comment (Ginxo) [reply .github/workflows/jpa_integration_ci.yml]: @rsynek applied to every `.yml` file

**Review by rsynek — APPROVED** [review]

**Pull request merged**